fast_api/types.py

```diff
@@ -0,0 +1,219 @@
+import json
+import datetime
+import decimal
+from typing import List
+from uuid import UUID
+
+
+class ExtendedRecord:
+    def __init__(self, **fields):
+        self._fields = fields
+
+    def get_record_data(self):
+        def alchemy_encoder(obj):
+            """Custom JSON encoder function for SQLAlchemy special types."""
+            if isinstance(obj, datetime.date):
+                return obj.isoformat()
+            elif isinstance(obj, decimal.Decimal):
+                return float(obj)
+            elif isinstance(obj, UUID):
+                return str(obj)
+
+        # Serializes the attributes in _fields using the custom encoder
+        return json.dumps(
+            {key: getattr(self, key) for key in self._fields}, default=alchemy_encoder
+        )
+
+
+class ExtendedSearch:
+    def __init__(
+        self,
+        sql: str = None,
+        query_limit: int = None,
+        query_offset: int = None,
+        full_count: int = None,
+        session_id: UUID = None,
```

Comment:
we seldom use UUID for ids. To double check => is this here intended?

Reply:
Yes, in a specific case, we have UserSessions returned and assigning its id to session_id in calls such as `resolve_records_by_static_slice` and `resolve_extended_search`. UserSessions is returning python UUIDs via the param as_uuid=True